ckan/tests/pytest_ckan/fixtures.py

```diff
@@ -248,3 +250,55 @@ def with_extended_cli(ckan_config, monkeypatch):
     # to apply per-test config changes to it without creating real
     # config file.
     monkeypatch.setattr(ckan.cli, u'load_config', lambda _: ckan_config)
+
+
+class FakeFileStorage(FlaskFileStorage):
+    content_type = None
+
+    def __init__(self, stream, filename):
+        self.stream = stream
+        self.filename = filename
+        self.name = u"upload"
+
+
+@pytest.fixture
+def make_resource(clean_db, ckan_config, monkeypatch, tmpdir):
```

Comment:
I'd call it `make_resource_with_upload` or `make_uploaded_resource`.
Or in fact what about `create_with_upload` and allow to pass a parameter for the upload field name. The defaults would be `action=resource_create` and `upload_field_name=upload` so you create a resource by default, but this way we can also use it to create users, orgs and groups with uploaded images.